sputnikdao-factory2/src/lib.rs

```diff
@@ -138,6 +144,109 @@ impl SputnikDAOFactory {
             .update_contract(account_id, code_hash, "update");
     }
 
+    /// Allows a DAO to store the official factory version as a blob, funded by the DAO wanting to upgrade
+    /// Required to successfully upgrade a DAO via proposals (proposal to store blob, proposal to upgrade from local blob)
+    /// Only intended for sputnik v2 DAO's created by sputnik factory
+    /// Payment is needed to cover storage costs for code blob size, paid by the DAO and returned upon blob removal
+    /// See docs for usage reference
+    #[payable]
+    pub fn store_contract_self(&mut self) {
+        let account_id = env::predecessor_account_id().as_bytes().to_vec();
+        let method_name = &"store_blob";
+
+        // Lock down contract upgrades to this factory:
+        let dao_id = env::predecessor_account_id().to_string();
+        let idx = dao_id.find('.').expect("INTERNAL_FAIL");
+        // ex: sputnik-dao.near
+        let factory_id = &dao_id[idx + 1..];
+
+        assert_eq!(factory_id, env::current_account_id().as_str(), "Wrong factory");
+
+        // Confirm payment before proceeding
+        assert!(
+            CODE_STORAGE_COST <= env::attached_deposit(),
+            "Must at least deposit {} to store",
+            CODE_STORAGE_COST
+        );
+
+        unsafe {
+            // Create a promise toward given account.
+            let promise_id =
+                sys::promise_batch_create(account_id.len() as _, account_id.as_ptr() as _);
+            sys::promise_batch_action_function_call(
+                promise_id,
+                method_name.len() as _,
+                method_name.as_ptr() as _,
+                DAO_CONTRACT_INITIAL_CODE.len() as _,
```

Comment:
This should be changed. The mainnet factory will not have `DAO_CONTRACT_INITIAL_CODE`.

We should fetch the code from the storage via a hash instead. The hash will be given as a parameter to the `store_contract_self` method.